src/sinks/redis.rs

```diff
@@ -0,0 +1,587 @@
+        if self.list_option.is_some() {
+            method = Option::from(self.list_option.unwrap().method)
+        }
+
+        let data_type = self.data_type;
+
+        let batch = BatchSettings::default()
+            .events(1)
+            .timeout(1)
+            .parse_config(self.batch)?;
+
+        let buffer = VecBuffer::new(batch.size);
+
+        let redis = RedisSink {
+            conn,
+            data_type,
+            method,
+        };
+
+        let svc = ServiceBuilder::new()
+            .settings(request, RedisRetryLogic)
+            .service(redis);
+
+        let sink = BatchSink::new(svc, buffer, batch.timeout, cx.acker())
+            .with_flat_map(move |e| stream::iter(encode_event(e, &key, &encoding)).map(Ok))
+            .sink_map_err(|error| error!(message = "Sink failed to flush.", %error));
+
+        Ok(super::VectorSink::Sink(Box::new(sink)))
+    }
+
+    async fn build_client(&self) -> RedisResult<ConnectionManager> {
+        trace!("Open Redis client.");
+        let client = redis::Client::open(self.url.as_str())?;
+        trace!("Open Redis client success.");
+        trace!("Get Redis connection.");
+        let conn = client.get_tokio_connection_manager().await;
+        trace!("Get Redis connection success.");
+        conn
+    }
+    async fn healthcheck(mut conn: ConnectionManager) -> crate::Result<()> {
+        redis::cmd("PING")
+            .query_async(&mut conn)
+            .await
+            .map_err(Into::into)
+    }
+}
+
+#[derive(Debug, Clone)]
+struct RedisKvEntry {
+    key: String,
+    value: Vec<u8>,
+}
+
+impl EncodedLength for RedisKvEntry {
+    fn encoded_length(&self) -> usize {
+        self.value.len()
+    }
+}
+
+fn encode_event(
+    mut event: Event,
+    key: &Template,
+    encoding: &EncodingConfig<Encoding>,
+) -> Option<EncodedEvent<RedisKvEntry>> {
+    let key = key
+        .render_string(&event)
+        .map_err(|error| {
+            emit!(TemplateRenderingFailed {
+                error,
+                field: Some("key"),
+                drop_event: true,
+            });
+        })
+        .ok()?;
+
+    encoding.apply_rules(&mut event);
+
+    let value = match encoding.codec() {
+        Encoding::Json => serde_json::to_vec(event.as_log())
+            .map_err(|error| panic!("Unable to encode into JSON: {}", error))
+            .unwrap_or_default(),
+        Encoding::Text => event
+            .as_log()
+            .get(log_schema().message_key())
+            .map(|v| v.as_bytes().to_vec())
+            .unwrap_or_default(),
+    };
+
+    let event = EncodedEvent::new(RedisKvEntry { key, value });
+    Some(event)
+}
+
+type RedisPipeResult = RedisResult<Vec<bool>>;
+
+impl Response for Vec<bool> {
+    fn is_successful(&self) -> bool {
+        let mut flag = true;
+        for r in self {
+            if !r {
+                flag = false;
+                break;
+            }
+        }
+        flag
```

Comment:
```suggestion
       self.iter().all(|x| x)
```

I think you could just write this as something like this.